packages/web-api-scan-runner/src/crawler/discovered-urls-processor.ts

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { injectable } from 'inversify';
+import _, { isNil } from 'lodash';
+import { WebsiteScanResult } from 'storage-documents';
+
+@injectable()
+export class DiscoveredUrlsProcessor {
+    public getProcessedUrls(discoveredUrls: string[], urlCrawlLimit: number, websiteScanResult?: WebsiteScanResult): string[] {
+        let processedUrls = discoveredUrls;
+        if (isNil(discoveredUrls)) {
+            return [];
+        }
+
+        if (!isNil(websiteScanResult?.knownPages)) {
+            processedUrls = this.removeUrlsFromList(discoveredUrls, websiteScanResult.knownPages);
+        }
+        processedUrls = this.limitNumUrls(processedUrls, this.getUrlLimit(urlCrawlLimit, websiteScanResult));
```

Comment:
You can get rid of the assignment here and just return the value of limitNumUrls.